server/src/main/java/io/spine/server/entity/RecordBasedRepository.java

```diff
@@ -162,16 +162,19 @@ public void store(Collection<E> entities) {
      */
     @Override
     public Optional<E> find(I id) {
-        Optional<EntityRecord> optional = findRecord(id);
-        if (!optional.isPresent()) {
-            return Optional.empty();
-        }
-        EntityRecord record = optional.get();
-        if (!record.isActive()) {
-            return Optional.empty();
-        }
-        E entity = toEntity(record);
-        return Optional.of(entity);
+        Optional<EntityRecord> record = findRecord(id);
+        Optional<E> result = record.filter(WithLifecycle::isActive)
+                                   .map(this::toEntity);
+        return result;
+    }
+
+    /**
+     * Finds an entity with the passed ID even if the entity is not
+     * {@linkplain WithLifecycle#isActive() active}.
+     */
+    public Optional<E> doFind(I id) {
```

Comment:
I would name this differently. An option coming to my mind is `findRaw`, meaning that no filtering is applied, as opposed to `find`, which filters by `archive == false && deleted == false`.